src/Promitor.Integrations.AzureMonitor/IAzureMonitorClient.cs

```diff
@@ -0,0 +1,13 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Promitor.Core.Metrics;
+
+namespace Promitor.Integrations.AzureMonitor
+{
+    public interface IAzureMonitorClient
```

Comment:
What do you think about using this interface to toggle between the new and legacy SDK? Unfortunately both implementations of this interface are still quite similar logically. Not sure how I'd abstract more though :/